src/components/conformance/utils.ts

```diff
@@ -7,10 +7,63 @@ import {
   TestOutcome,
   TestResult,
   TestStats,
+  UrlState,
   VersionedStats,
   VersionItem,
 } from "@site/src/components/conformance/types";
 
+// Take a search param and create a state object
+export function createUrlState(search: string): UrlState {
+  const params = new URLSearchParams(search);
+  console.log(`testPath: ${params.get("testPath")}`);
+  return {
+    versionTag: params.get("version") ?? undefined,
+    testPath: params.get("testPath")?.split("/") ?? undefined,
+    selectedTest: params.get("selectedTest") ?? undefined,
+  };
+}
+
+export function updateInitialConformanceState(
+  urlState: UrlState,
+  conformanceState: ConformanceState,
+) {
+  if (
+    !conformanceState &&
+    (urlState.versionTag || urlState.testPath || urlState.selectedTest)
+  ) {
+    const fetchUrl =
+      urlState.versionTag === "main"
+        ? `https://raw.githubusercontent.com/boa-dev/data/main/test262/refs/heads/main/latest.json`
+        : `https://raw.githubusercontent.com/boa-dev/data/main/test262/refs/tags/${urlState.versionTag}/latest.json`;
+
+    const testPath = urlState.testPath || [];
+    return {
+      version: { tagName: urlState.versionTag, fetchUrl },
+      testPath: [urlState.versionTag, ...testPath],
+      ecmaScriptVersion: undefined,
+      sortOption: availableSortingOptions[0].id,
+      selectedTest: urlState.selectedTest,
+    };
+  }
+  return conformanceState;
+}
+
+export function createSearchParams(
+  version: VersionItem,
+  testPath?: string[],
+  selectedTest?: string,
+) {
+  const search = new URLSearchParams();
+  search.append("version", version.tagName);
+  if (testPath) {
```

Comment:
issue: It is possible here for `testPath` to be ["main"].

This would then make `testPath`'s value an empty string.

```suggestion
  if (testPath && testPath.length > 1) {
```